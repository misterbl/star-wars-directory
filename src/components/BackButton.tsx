import React, { PureComponent } from "react";

interface IBackButton {
  route: string;
  text: string;
  push: (path: string, state?: any) => void;
  onClick?: () => void;
}
class BackButton extends PureComponent<IBackButton> {
  pushToPage = () => {
    const { push, route, onClick } = this.props;
    onClick && onClick();
    push(route);
  };
  render() {
    return (
      <div className="btn btn-dark mt-3 ml-3 pl-4" onClick={this.pushToPage}>
        <div className=" back-button" />
        {this.props.text}
      </div>
    );
  }
}

export default BackButton;
