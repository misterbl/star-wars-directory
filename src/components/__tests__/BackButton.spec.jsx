/* eslint-disable no-undef */
import * as React from "react";
import { shallow } from "enzyme";
import BackButton from "../BackButton";

describe("BackButton", () => {
  const props = {
    route: "route",
    text: "text",
    push: jest.fn()
  };
  const wrapper = shallow(<BackButton {...props} />);
  it("matches the snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });
  it("calls push() onClick ", () => {
    const button = wrapper.find("button").at(0);
    button.simulate("click");
    expect(props.push).toHaveBeenCalled();
  });
});
