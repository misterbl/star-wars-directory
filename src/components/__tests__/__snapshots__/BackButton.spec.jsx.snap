// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BackButton matches the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <BackButton
    push={[MockFunction]}
    route="route"
    text="text"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="back-button"
        />,
        "text",
      ],
      "className": "btn btn-dark mt-3 ml-3 pl-4",
      "onClick": [Function],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "back-button",
        },
        "ref": null,
        "rendered": null,
        "type": "div",
      },
      "text",
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="back-button"
          />,
          "text",
        ],
        "className": "btn btn-dark mt-3 ml-3 pl-4",
        "onClick": [Function],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "back-button",
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
        "text",
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
